unlock-app/src/middlewares/storageMiddleware.js

```diff
@@ -18,15 +17,21 @@ import {
   gotEncryptedPrivateKeyPayload,
 } from '../actions/user'
 import UnlockUser from '../structured_data/unlockUser'
+import Error from '../utils/Error'
+import { setError } from '../actions/error'
+
+const { Storage } = Error
 
 const storageMiddleware = config => {
   const { services } = config
   return ({ getState, dispatch }) => {
     const storageService = new StorageService(services.storage.host)
 
     // NEW_TRANSACTION
-    storageService.on(failure.storeTransaction, error => {
-      dispatch(storageError(error))
+    storageService.on(failure.storeTransaction, () => {
+      // TODO: we are in control of what storageService emits --
+      // construct helpful errors at source of failure?
```

Comment:
Right now many of these errors are constant, because digging through the services and making sure they all return a useful error message was a bit out of scope for this PR. Future work can include making sure we construct useful messages to emit, and then the diagnostics can be more specific.